invenio_search/ext.py

```diff
@@ -64,12 +63,18 @@ def __init__(self, app,
 
     @cached_property
     def templates(self):
-        result = None
+        """Generate a dictionary with template names and file paths."""
+        templates = {}
+        result = []
         if self.entry_point_group_templates:
             result = self.load_entry_point_group_templates(
-                self.entry_point_group_templates)
-        return {k: v for d in result for k, v in d.items()} \
-            if result is not None else {}
+                self.entry_point_group_templates) or []
+
+        for template in result:
+            for name, path in template.items():
```

Comment:
``six.iteritems`` if we care about ``2`` efficiency etc

Reply:
We are going to drop Python 2 support by the end of the year, it will be in one of the next Invenio-Sprints. WDYT?